packages/kbn-alerts-ui-shared/src/rule_type_modal/components/index.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo, useState } from 'react';
+import type { HttpStart } from '@kbn/core-http-browser';
+import type { ToastsStart } from '@kbn/core-notifications-browser';
+import { useLoadRuleTypesQuery } from '../../common/hooks';
+import { RuleTypeModal, type RuleTypeModalProps } from './rule_type_modal';
+import { filterAndCountRuleTypes } from './helpers/filter_and_count_rule_types';
+
+export interface RuleTypeModalComponentProps
+  extends Pick<RuleTypeModalProps, 'onClose' | 'onSelectRuleType'> {
```

Comment:
nitpick: not a huge fan of extending types if we just need 2 of the props.

suggestion:

```
export interface RuleTypeModalComponentProps {
  onClose: RuleTypeModalProps['onClose'];
  onSelectRuleType: RuleTypeModalProps['onSelectRuleType'];
  ...
}
```